Add unit tests to enhance code coverage per issue #2

mdg357 opened the pull request:
Achieve 100% code coverage on Model.Evolution and Evolution.Logic

mdg357 committed:
Add unit tests to enhance code coverage per issue #2

Achieve 100% code coverage on Model.Evolution and Evolution.Logic

DillonAd commented:
Isn't there a possibility of conflicting behaviors with the mocks being shared between the tests? I would think that if two tests were running in parallel with the same behavior getting set up two different ways it might cause random failures.

Would it make more sense for the mocked objects to be local to the test they are being used in?

mdg357 commented:
That's a good point. It's not an issue yet, simply because there aren't a lot of tests, but I could see how it might cause issues as things become more complex. I'll make the change.

Commit:
Switch Mock objects to be local variables

Depending on execution order or parallelism used, different behaviors are possible for the Mocked objects. Use local instances instead to mitigate this issue.

DillonAd commented:
Looks great! Thank you!
